test/utils/base/cluster_context.go

```diff
@@ -148,9 +148,76 @@ func (cc *ClusterContext) DumpTestInfo(dirName string) {
 	} else {
 		log.Printf("error dumping test info: %v", err)
 	}
-	out, err := cc.KubectlExec("get -o wide job,pod,service")
+
+	log.Printf("namespace status:")
+	_, err = cc.KubectlExec("get -o wide job,pod,service,event")
 	if err != nil {
 		log.Printf("failed getting kube info: %v", err)
 	}
-	log.Printf("kube info: \n%v", string(out))
+
+	// These may be up and running now (and their output will show that).  However,
+	// the last time they _terminated_, it was with a non-zero return, and that
+	// may be valuable information for investigations.
+	//
+	// Here, we first get their pod and container names, and show some debugging
+	// information...
+	log.Printf("pod container whose last termination was non-zero:")
+	out, err := cc.KubectlExec(`get pod -o go-template='
```

Comment:
Why don't you simply use the k8s api directly to list and filters pod?

Reply:
I started with `kubectl` and never looked back? :)

Everything else was doing `kubectl`, so I just used it here as well.  As an advantage of this approach, this (huge) Go template stands out on the logs, allowing others to understand what's going on and reuse it manually while doing investigations, straight from the log file.

The only disadvantage I see is that it is kinda ugly :)  I tried jsonpath, but I don't think it can accomplish what I need here.

Let me know if you have some strong position on this.  As primarily a log reader, I prefer it this way.  But I have no issues in changing it.

Reply:
Not an issue for now, but I think using the k8s api would make it simpler and easier to read.